File: examples/python/asticaListen_sample.py
import requests
import json
import base64
import os
def get_audio_base64_encoding(audio_path: str) -> str:
    """
    Function to return the base64 string representation of an audio file
    """
    with open(audio_path, 'rb') as file:
        audio_data = file.read()
    audio_extension = os.path.splitext(audio_path)[1]
    base64_encoded = base64.b64encode(audio_data).decode('utf-8')
    #Make sure to append base64 header
    return f"data:audio/{audio_extension[1:]};base64,{base64_encoded}"
    
def asticaAPI(endpoint, payload, timeout):
    response = requests.post(endpoint, data=json.dumps(payload), timeout=timeout, headers={ 'Content-Type': 'application/json', })
    if response.status_code == 200:
        return response.json()
    else:
        return {'status': 'error', 'error': 'Failed to connect to the API.'}

asticaAPI_key = 'YOUR API KEY' # visit https://astica.ai
asticaAPI_timeout = 25 # seconds
asticaAPI_endpoint = 'https://listen.astica.ai/transcribe'
asticaAPI_modelVersion = '1.0_full'

asticaAPI_doStream = 0 # Determines whether to display responses in real-time.
asticaAPI_low_priority = 0 # Lower costs by receiving a URL to query for results. 

if 1 == 1:
    asticaAPI_input = 'https://astica.ai/example/asticaListen_sample.wav' # use https audio input (faster)
else:
    asticaAPI_input = get_audio_base64_encoding('input.wav')  # use base64 audio input (slower)

    
# Define payload dictionary
asticaAPI_payload = {
    'tkn': asticaAPI_key,
    'modelVersion': asticaAPI_modelVersion,
    'input': asticaAPI_input,
    'doStream': asticaAPI_doStream,
    'low_priority': asticaAPI_low_priority
}


# Call API function and store result
asticaAPI_result = asticaAPI(asticaAPI_endpoint, asticaAPI_payload, asticaAPI_timeout)

print('<hr><b>astica API Output:</b><br>')
print(json.dumps(asticaAPI_result, indent=4))

#Handle asticaAPI response
if 'status' in asticaAPI_result:
    # Output Error if exists    
    if asticaAPI_result['status'] == 'error':
        print('Output:', asticaAPI_result['error'])
    # Output Success if exists
    if asticaAPI_result['status'] == 'success':
        if 'resultURI' in asticaAPI_result:
            print('Output URI:', asticaAPI_result['resultURI'])
            print('Query this URL to obtain the output of your results')
        else:
            print('Output:', asticaAPI_result['text'])
else:
    print('Invalid response')

File: examples/node/asticaListen_sample.js
const axios = require('axios');
const fs = require('fs');

const asticaAPI_key = 'YOUR API KEY';
const asticaAPI_timeout = 15;

const asticaAPI_endpoint = 'https://listen.astica.ai/transcribe';
const asticaAPI_modelVersion = '1.0_full';
const asticaAPI_doStream = 0;
const asticaAPI_low_priority = 0;

//Input Method 1: https URL of a jpg/png image (faster)
var asticaAPI_input = 'https://astica.ai/example/asticaListen_sample.wav';

/*
    //Input Method 2: base64 encoded string of a local image (slower)  
    var path_to_local_file = 'input.wav';
    var audio_data = fs.readFileSync(path_to_local_file);
    var audio_extension = path_to_local_file.split('.').pop();
    //For now, let's make sure to prepend appropriately with: "data:image/extension_here;base64" 
    var asticaAPI_input = `data:audio/${audio_extension};base64,${audio_data.toString('base64')}`;
*/

//Prepare payload
const asticaAPI_payload = {
    tkn: asticaAPI_key,
    modelVersion: asticaAPI_modelVersion,
    input: asticaAPI_input,
    doStream: asticaAPI_doStream,
    low_priority: asticaAPI_low_priority
};

//console.log(asticaAPI_payload);
//Demo function
(async () => {    
    // Submit request
    const result = await asticaAPI(asticaAPI_endpoint, asticaAPI_payload, asticaAPI_timeout);    
    console.log('astica API Output:\n', JSON.stringify(result, null, 4));
    
    // Handle response
    if (result.status === 'error') {
        console.log('Error:', result.error);
    } else if (result.status === 'success') {
        if (result.resultURI) {
            console.log('Output URI:', result.resultURI, '\nQuery this URL to obtain the output of your results');
        } else {
            console.log('Output:', result.text);
        }
    } else { console.log('Invalid response'); }
    
    
})();



// Required astica api function
async function asticaAPI(endpoint, payload, timeout) {
    try {
        const response = await axios.post(endpoint, payload, {
            headers: { 'Content-Type': 'application/json' },
            timeout: timeout * 1000, // timeout in ms
        });

        if (response.status === 200) {
            return response.data;
        }
    } catch (error) {
        return { status: 'error', error: 'Failed to connect to the API.' };
    }
}